newrelic_plugin_agent/plugins/couchbase.py

```diff
@@ -0,0 +1,158 @@
+        {'type': 'nodes', 'label': 'interestingStats.couch_views_actual_disk_size', 'suffix': 'byte'},
+        {'type': 'nodes', 'label': 'interestingStats.couch_views_data_size', 'suffix': 'byte'},
+        {'type': 'nodes', 'label': 'interestingStats.mem_used', 'suffix': 'byte'},
+        {'type': 'nodes', 'label': 'interestingStats.ops', 'suffix': 'count'},
+        {'type': 'nodes', 'label': 'interestingStats.curr_items', 'suffix': 'count'},
+        {'type': 'nodes', 'label': 'interestingStats.curr_items_tot', 'suffix': 'count'},
+        {'type': 'nodes', 'label': 'interestingStats.vb_replica_curr_items', 'suffix': 'count'},
+
+        {'type': 'nodes', 'scoreboard': True, 'score_value': 'active', 'label': 'clusterMembership', 'suffix': 'count'},
+        {'type': 'nodes', 'scoreboard': True, 'score_value': 'healthy', 'label': 'status', 'suffix': 'count'},
+
+        {'type': 'buckets', 'label': 'basicStats.quotaPercentUsed', 'suffix': 'percent'},
+        {'type': 'buckets', 'label': 'basicStats.opsPerSec', 'suffix': 'ops'},
+        {'type': 'buckets', 'label': 'basicStats.diskFetches', 'suffix': 'percent'},
+        {'type': 'buckets', 'label': 'basicStats.itemCount', 'suffix': 'percent'},
+        {'type': 'buckets', 'label': 'basicStats.diskUsed', 'suffix': 'byte'},
+        {'type': 'buckets', 'label': 'basicStats.dataUsed', 'suffix': 'byte'},
+        {'type': 'buckets', 'label': 'basicStats.memUsed', 'suffix': 'byte'},
+    ]
+
+    def add_datapoints(self, data):
+        """Add data points for all couchbase nodes.
+
+        :param dict stats: stats for all nodes
+
+        """
+        # fetch metrics for each metric type (cluster, nodes, buckets)
+        for typ, stats in data.iteritems():
+            for metric in [m for m in self.METRICS if m['type'] == typ]:
+                if typ == 'cluster':
+                    items = stats
+                    name_key = 'name'
+                elif typ == 'nodes':
+                    items = stats['nodes']
+                    name_key = 'hostname'
+                elif typ == 'buckets':
+                    items = stats
+                    name_key = 'name'
+
+                # count score for scoreboard metrics,
+                # otherwise just add the gauge value from the API repsonse
+                if metric.get('scoreboard', False):
+                    self.add_gauge_value(
+                        '%s/_scoreboard/%s' % (typ, metric['label']),
+                        metric['suffix'],
+                        self._get_metric_score(metric, items))
+                else:
+                    # add gauge value for current metric.
+                    # NOTE cluster metrics are not repeated,
+                    # as there is a single cluster.
+                    # nodes and bucket metrics are repeated,
+                    # as there are (usually) several of them
+                    if typ == 'cluster':
```

Comment:
There are quite a few branching in this for loop. maybe it would be cleaner to have a for loop per `typ`